### PR TITLE
Add hdf5-vol-async package.

### PR DESCRIPTION
**Pull request by lrknox**
Add HDF5 1.13.0-rc6 version for building vol-async.
1st step in proposed plan from PR #26305

### CODE REVIEW

**Review by hyoklee — APPROVED** [review]

**Review by byrnHDF — DISMISSED** [review]

File: var/spack/repos/builtin/packages/hdf5-vol-async/package.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+class Hdf5VolAsync(CMakePackage):
+    """This package enables asynchronous IO in HDF5."""
+
+    homepage = "https://sdm.lbl.gov/"
+    git      = "https://github.com/hpc-io/vol-async"
+    maintainers = ['hyoklee']
+
+    version('v1.0')
+    depends_on('argobots@main')
+    depends_on('hdf5@1.13.0-rc6+mpi+threadsafe')
```
Comment (michaelkuhn) [inline var/spack/repos/builtin/packages/hdf5-vol-async/package.py]: We typically do not add pre-release snapshots to Spack. How about doing `depends_on('hdf5@develop-1.13:+mpi+threadsafe')`?

Comment (lrknox) [reply var/spack/repos/builtin/packages/hdf5-vol-async/package.py]: I'm fine with that - maybe I misinterpreted a comment in the earlier draft PR.  I'll plan to make that change.

Comment (sethrj) [reply var/spack/repos/builtin/packages/hdf5-vol-async/package.py]: Part of the reason is that with the spack versioning scheme, `Version('1.13.0-rc6') > Version('1.13.0')` so the concretizer will prefer the prerelease to the actual release....

File: var/spack/repos/builtin/packages/hdf5-vol-async/package.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    def cmake_args(self):
+        """Populate cmake arguments for HDF5 VOL."""
+        args = [
+            '-DBUILD_SHARED_LIBS:BOOL=ON',
+            '-DBUILD_TESTING:BOOL=ON'
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/hdf5-vol-async/package.py]:
```suggestion
            self.define('BUILD_SHARED_LIBS', True),
            self.define('BUILD_TESTING:BOOL', True),
```

Also don't you usually want to build testing only when `self.run_tests` or doing `spack develop`?

**Review by hyoklee — DISMISSED** [review]

File: var/spack/repos/builtin/packages/hdf5-vol-async/package.py (hunk 1 above)
Comment (lrknox) [reply var/spack/repos/builtin/packages/hdf5-vol-async/package.py]: Changed hdf5 version to develop-1.13

File: var/spack/repos/builtin/packages/hdf5-vol-async/package.py (hunk 2 above)
Comment (lrknox) [reply var/spack/repos/builtin/packages/hdf5-vol-async/package.py]: Updated according to both suggestions.

**Review by hyoklee — APPROVED** [review]

**Review by chuckatkins — APPROVED** [review]

**Pull request merged**